src/opensynth/models/faraday/gaussian_mixture/model.py

```diff
@@ -194,8 +194,14 @@ def sample(self, num_datapoints: int) -> torch.Tensor:
             datapoints, call this method multiple times.
         """
         # First, we sample counts for each
+
+        # Normalise component probabilities after converting to float64
+        # This is necessary because numpy casts the probabilities to float64
+        # and we want to avoid precision issues resulting from this conversion
+        # See https://github.com/numpy/numpy/issues/8317.
+        _probs = self.component_probs.numpy().astype("float64")
         component_counts = np.random.multinomial(
-            num_datapoints, self.component_probs.numpy()
+            num_datapoints, _probs / _probs.sum()
         )
```

Comment:
Add a check to ensure normalised component probabilities sum to one (raise Error if not).

Reply:
We could definitely do that, but the error message from the np.random.multinomial(..., _probs / _probs.sum() )  is `ValueError: sum(pvals[:-1]) > 1.0` which I think is fairly self explanatory? Adding our own error handling is of course fine, but I wasn't sure if it's necessary. What do you think? 

also by definition `_probs / _probs.sum()` will sum to 1, unless there's issues with floating point precision (thats why we convert to float64 before doing this or if all `_probs` are zero, which I think would have triggered errors earlier in the code.)